### PR TITLE
[SPARK-29491][SQL] Add bit_count function support

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

BIT_COUNT(N) - Returns the number of bits that are set in the argument N as an unsigned 64-bit integer, or NULL if the argument is NULL

### Why are the changes needed?

Supported by MySQL，Microsoft SQL Server ，etc.

### Does this PR introduce any user-facing change?

add a built-in function
### How was this patch tested?

add uts

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan @wangyum

File: sql/core/src/test/resources/sql-tests/inputs/bitwise.sql [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+-- negative num
+select bit_count(-1L);
+
+-- mysql> select 9223372036854775807 + 1;
+-- ERROR 1690 (22003): BIGINT value is out of range in '(9223372036854775807 + 1)'
+-- mysql> select bit_count(9223372036854775807 + 1);
+-- ERROR 1690 (22003): BIGINT value is out of range in '(9223372036854775807 + 1)'
+-- out of range
+select 9223372036854775807L + 1L;
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/inputs/bitwise.sql]: the overflow behavior is nothing to do with the `bit_count` function, or did I miss something?

File: sql/core/src/test/resources/sql-tests/inputs/bitwise.sql [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+select bit_count(9223372036854775807L + 1L);
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/inputs/bitwise.sql]: we can test the min long value if that's what you intend to do here.

File: sql/core/src/test/resources/sql-tests/inputs/bitwise.sql (hunk 1 above)
Comment (yaooqinn) [reply sql/core/src/test/resources/sql-tests/inputs/bitwise.sql]: yes, you're right. The case here illustrates the overflow behavior is nothing to do with the `bit_count`, which may cause inconsistency with mysql

File: sql/core/src/test/resources/sql-tests/inputs/bitwise.sql (hunk 2 above)
Comment (yaooqinn) [reply sql/core/src/test/resources/sql-tests/inputs/bitwise.sql]:
```mysql
mysql> select bit_count(-9223372036854775808);
+---------------------------------+
| bit_count(-9223372036854775808) |
+---------------------------------+
|                               1 |
+---------------------------------+
1 row in set (0.01 sec)

mysql> select bit_count(9223372036854775807);
+--------------------------------+
| bit_count(9223372036854775807) |
+--------------------------------+
|                             63 |
+--------------------------------+
1 row in set (0.00 sec)

mysql> select bit_count(9223372036854775807 + 1);
ERROR 1690 (22003): BIGINT value is out of range in '(9223372036854775807 + 1)'
```

File: sql/core/src/test/resources/sql-tests/inputs/bitwise.sql (hunk 1 above)
Comment (cloud-fan) [reply sql/core/src/test/resources/sql-tests/inputs/bitwise.sql]: We have test suites for the overflow behavior. I don't think it makes sense to test overflow again in all the functions that take integers.

Comment (yaooqinn) [reply sql/core/src/test/resources/sql-tests/inputs/bitwise.sql]: ok, I will rm these and just check long min/max

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**